### PR TITLE
Add Online Inference notebooks for ASR and Speech Commands

### PR DESCRIPTION
**Pull request by titu1994**
# Changelog
- Adds online inference notebook ported to PTL
- Adds support for pretrained checkpoints for MatchboxNet

# Note
Due to rebase, speech commands notebook (prev number 2, now number 3) is showing up as different notebooks but its the same as before that @blisc reviewed the past week.

### CODE REVIEW

**Comment by okuchaiev** [comment]
setting dither to 0.00001 seems to improve transcripts. @vsl9 could you please have a look? any other ideas for improvements?

**Comment by titu1994** [comment]
> where did spectrogram normalization constants came from?
From the Nemo 0.11x notebook for Online inference. 

> setting dither to 0.00001 seems to improve transcripts
Hmm, odd. Maybe cause im evaluating on a train_clean sample so it overfit to that?

> 04_Online_Speech_Commands_Demo just continuously outputs classes (even when I don't say anything. Online ASR demo doesn't have this problem). Is there None or silence class?

No, this is a direct 35 way classes (no baseline class). The 10+2 class subset does support background classes but we dont have checkpoints on NGC for that.

**Comment by okuchaiev** [comment]
"04_Online_Speech_Commands_Demo" really isn't functional with microphone - when I don't say anything it just outputs classes.
Any ideas to disable classification when sound signal is smaller than something? @vsl9

**Comment by titu1994** [comment]
@okuchaiev Changed framerate to 1.0 for ASR demo 2 and changed online inference model for MBN to 10+2 model.

**Comment by okuchaiev** [comment]
I don't see changes??

**Comment by titu1994** [comment]
Hmm stuff got reverted for some reason. Dunno what happened

**Comment by titu1994** [comment]
Seems I botched an earlier rebase, should be fine now @okuchaiev

**Comment by okuchaiev** [comment]
@titu1994 I think there are two options (you chose):

[1] Merge "02_Online_ASR_Microphone_Demo" without Online_Speech_Commands. Work on Online_Speech_Commands in separate PR
[2] Work on Online_Speech_Commands as part of this PR.

The main problem with Online_Speech_Commands tutorial is that keeps triggering when I am not speaking into the microphone at all. So in order for this to work in *streaming* mode it really needs "silence" label which it does not have.
Thus two ways of fixing it:
(a) add "silence" label - but this will require re-training
(b) Make it into non-streaming demo. E.g. require a user to execute a cell to record  a command then execute another one for recognizing.

**Comment by titu1994** [comment]
@okuchaiev That shouldn't happen, im not printing the last two labels at all (silence and background). Are you using the 12-class version here https://github.com/NVIDIA/NeMo/blob/85951047b78db45fadbebf977ce69b1477de2cee/tutorials/asr/04_Online_Speech_Commands_Demo.ipynb?

 The last two classes are suppressed using 

```python
    @staticmethod
    def _greedy_decoder(logits, vocab):
        s = ''
        
        if logits.shape[0]:
            class_idx = np.argmax(logits)
            class_label = vocab[class_idx]
            if class_idx < 10:  # <--- this line here, skips "silence" and "background classes [10 and 11]
                s += str(class_label) + "\n"
        return s
```

**Comment by okuchaiev** [comment]
Here is example output when I say nothing to the microphone for some time.
(I made it print silence tag)

```
Available audio input devices:
4 HD-Audio Generic: ALCS1200A Analog (hw:1,0)
6 HD-Audio Generic: ALCS1200A Alt Analog (hw:1,2)
7 Logitech USB Headset: Audio (hw:2,0)
8 C922 Pro Stream Webcam: USB Audio (hw:3,0)
10 pulse
11 default
Please type input device ID:
7
Listening...
silence
up
silence
up
silence
silence
silence
silence
up
up
up
off
up
up
up
up
up
off
up
off
up
off
up
up
up
up
up
up
up
unknown
up
up
up
up
up
unknown
up
off
off
up
off
off
up
off
up
off
off
off
up
up
off
off
off
off
off
up
unknown
unknown
off
up
```

**Comment by titu1994** [comment]
Hmm that's not something that can be easily fixed. Maybe it's buffer size is too small (0.25) but it could also be that model was trained on silence that doesn't correspond to real silence (the dataset uses 4-5 files of silence strides by certain small window to generate it's silence set). 

I think for now I'll remove the speech command notebook from online ASR and submit another notebook. But the fix to this is non trivial.

**Comment by titu1994** [comment]
@okuchaiev Removed the Speech command notebook, and put it in a wip PR for now.

**Review by okuchaiev — APPROVED** [review]
